src/common/util/array-move.ts

```diff
@@ -20,14 +20,29 @@ function findNestedItem(
   }, obj);
 }
 
+function updateNestedItem(obj: any, path: ItemPath): any {
+  const lastKey = path.pop()!;
+  const parent = findNestedItem(obj, path);
+  parent[lastKey] = [...parent[lastKey]];
+  return obj;
+}
+
 export function nestedArrayMove<A>(
   obj: A,
   oldIndex: number,
   newIndex: number,
   oldPath?: ItemPath,
   newPath?: ItemPath
 ): A {
-  const newObj = (Array.isArray(obj) ? [...obj] : { ...obj }) as A;
+  let newObj = (Array.isArray(obj) ? [...obj] : { ...obj }) as A;
+
+  if (oldPath) {
+    newObj = updateNestedItem(newObj, [...oldPath]);
+  }
+  if (newPath) {
+    newObj = updateNestedItem(newObj, [...newPath]);
+  }
+
   const from = oldPath ? findNestedItem(newObj, oldPath) : newObj;
   const to = newPath ? findNestedItem(newObj, newPath, true) : newObj;
```

Comment:
We already have `from` and `to` here right, we can just make a copy here?

Reply:
Or use `toSpliced` instead of `splice` (not supported on older browsers)

Reply:
We have to set a new reference for `from` and `to` and we have to include this into newObj.

I set the new reference before manipulating, because there can be the case of index shifting which makes it way more complex afterwards.
Then I get the referenced `from` and `to` from new object and manipulate them. 

If I say `from = [...findNestedItem(newObj, oldPath)]`, from wouldn't update inside newObj anymore and I would need to include it at the correct position again.

toSpliced isn't well supported as you said and we woulnd't get a benefit out of it, because we would need to insert the spliced array again into the parent object.

 I also tried to clone the full object, but then we collapse everything because this is bound to the reference too.


If you find a simpler solution please suggest :slightly_smiling_face: